### PR TITLE
A task to search and report TODOs associated with closed issues

### PR DESCRIPTION
**Pull request by dvoytenko**

### CODE REVIEW

File: build-system/tasks/todos.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+ * @param {!File} file file is a vinyl file object
+ * @param {string} issueId
+ * @param {string} todo
+ * @return {!Promise<number>}
+ */
+function reportClosedIssue(file, issueId, todo) {
+  if (issueCache[issueId] !== undefined) {
+    return Promise.resolve(issueCache[issueId]);
+  }
+  return githubRequest('/issues/' + issueId).then(response => {
```
Comment (jridgewell) [inline build-system/tasks/todos.js]: This'll make multiple requests to Github. Instead, you could cache this promise (instead of its resoled value).

Comment (dvoytenko) [inline build-system/tasks/todos.js]: Good point. Will do.

Comment (dvoytenko) [inline build-system/tasks/todos.js]: Done

File: build-system/tasks/todos.js [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+var path = require('path');
+var srcGlobs = require('../config').presubmitGlobs;
+var util = require('gulp-util');
+var through2 = require('through2');
+var request = BBPromise.promisify(require('request'));
+
+var GITHUB_ACCESS_TOKEN = process.env.GITHUB_ACCESS_TOKEN;
+
+/** @type {!Object<string, !Promise<number>>} */
+var issueCache = {};
```
Comment (erwinmombay) [inline build-system/tasks/todos.js]: Object.create

**Comment by erwinmombay** [comment]
@dvoytenko can you add this to https://github.com/ampproject/amphtml/blob/master/contributing/DEVELOPING.md#build--test

File: build-system/tasks/todos.js (hunk 2 above)
Comment (dvoytenko) [inline build-system/tasks/todos.js]: Done

**Comment by dvoytenko** [comment]
@erwinmombay Added to `DEVELOPMENT.md`

**Pull request merged**